android_jni/avifandroidjni/src/main/jni/libavif_jni.cc

```diff
@@ -52,6 +52,7 @@ bool CreateDecoderAndParse(AvifDecoderWrapper* const decoder,
   }
   decoder->decoder->ignoreXMP = AVIF_TRUE;
   decoder->decoder->ignoreExif = AVIF_TRUE;
+  decoder->decoder->strictFlags = AVIF_STRICT_DISABLED;
```

Comment:
It is better to disable specific flags. Chrome's third_party/blink/renderer/platform/image-decoders/avif/avif_image_decoder.cc has the following:

```
    // Turn off libavif's 'clap' (clean aperture) property validation. (We
    // ignore the 'clap' property.)
    decoder_->strictFlags &= ~AVIF_STRICT_CLAP_VALID;
    // Allow the PixelInformationProperty ('pixi') to be missing in AV1 image
    // items. libheif v1.11.0 or older does not add the 'pixi' item property to
    // AV1 image items. (This issue has been corrected in libheif v1.12.0.) See
    // crbug.com/1198455.
    decoder_->strictFlags &= ~AVIF_STRICT_PIXI_REQUIRED;
```

Pleae see if you can get by with only `decoder->decoder->strictFlags &= ~AVIF_STRICT_PIXI_REQUIRED;`

Reply:
The current android platform's AVIF parser is quite permissive and does not care about most errors (it only cares about those boxes that are actually necessary to decode the image).

So it is better for the android JNI wrapper to try and match that as much as possible. So turning off strict mode altogether is probably better.

Reply:
The libavif decoder performs a lot of standard compliance checks unconditionally. It is impossible for the Android JNI wrapper to match the permissiveness of the Android platform's AVIF parser. It's better to match the strictness of Chrome and help improve standard compliance of AVIF files.

Reply:
> The libavif decoder performs a lot of standard compliance checks unconditionally. It is impossible for the Android JNI wrapper to match the permissiveness of the Android platform's AVIF parser. 

Hmm, the whole point of bundling libavif along with an Android app is to make sure that avif decoding is available irrespective of the android platform version. it is also generally considered to be a super-set of what the android platform supports (like 10bit, 12bit, etc). So it would be weird if an avif file can be parsed and rendered by the platform but the libavif wrapper fails to display it because it failed some compliance checks.

> It's better to match the strictness of Chrome and help improve standard compliance of AVIF files.

Also in general, i believe it is not fair to enforce conformance checks on decoders in order to improve compliance. Because once such files are in the wild, they are in the wild and we should not go out of our way to fail on them especially if we can decode them.

Reply:
The decision to add standard compliance checks to libavif was supported by several members of the AOM Still Image Working Group. The rationale is that it will help prevent non-standard-compliant AVIF files from being created, and we can only do this before too many non-compliant AVIF files have been created.

Based on Chrome's AVIF bug reports, I recommend only disabling `AVIF_STRICT_PIXI_REQUIRED`, but ultimately this is your call. My main request is to disable specific strict flags instead of `AVIF_STRICT_DISABLED`, with comments that justify why a strict flag is disabled. For example, this is how Chrome disables two strict flags:

```
    // Turn off libavif's 'clap' (clean aperture) property validation. (We
    // ignore the 'clap' property.)
    decoder_->strictFlags &= ~AVIF_STRICT_CLAP_VALID;
    // Allow the PixelInformationProperty ('pixi') to be missing in AV1 image
    // items. libheif v1.11.0 or older does not add the 'pixi' item property to
    // AV1 image items. (This issue has been corrected in libheif v1.12.0.) See
    // crbug.com/1198455.
    decoder_->strictFlags &= ~AVIF_STRICT_PIXI_REQUIRED;
```

Note: We may turn on `AVIF_STRICT_CLAP_VALID` in Chrome when we support 'clap' (likely with restrictions).

Reply:
> The decision to add standard compliance checks to libavif was supported by several members of the AOM Still Image Working Group.

Yes i understand that from an encoder perspective. We should strive as much as possible to not create invalid AVIF files. But it is impossible for us to control the entire ecosystem (there will always be that one popular tool that is stuck on an older version of ffmpeg or something that will produce slightly invalid AVIF files). Ultimately, i still feel that if we can decode a file, we should not go out of our way to bail out.

> The rationale is that it will help prevent non-standard-compliant AVIF files from being created, and we can only do this before too many non-compliant AVIF files have been created.
> 
Some of the files in the "official" AV1 github samples fail the CLAP check: https://github.com/AOMediaCodec/av1-avif/tree/master/testFiles/Link-U (kimono.crop.avif for example). So such files are already in the wild. :)

> Based on Chrome's AVIF bug reports, I recommend only disabling `AVIF_STRICT_PIXI_REQUIRED`, but ultimately this is your call. My main request is to disable specific strict flags instead of `AVIF_STRICT_DISABLED`, with comments that justify why a strict flag is disabled. For example, this is how Chrome disables two strict flags:
> 
> ```
>     // Turn off libavif's 'clap' (clean aperture) property validation. (We
>     // ignore the 'clap' property.)
>     decoder_->strictFlags &= ~AVIF_STRICT_CLAP_VALID;
>     // Allow the PixelInformationProperty ('pixi') to be missing in AV1 image
>     // items. libheif v1.11.0 or older does not add the 'pixi' item property to
>     // AV1 image items. (This issue has been corrected in libheif v1.12.0.) See
>     // crbug.com/1198455.
>     decoder_->strictFlags &= ~AVIF_STRICT_PIXI_REQUIRED;
> ```
> 
> Note: We may turn on `AVIF_STRICT_CLAP_VALID` in Chrome when we support 'clap' (likely with restrictions).

I have copied the specific flags from chromium in this CL. PTAL.